src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (isEmpty()) {
+            Node<T> node = new Node<>(value, null, null);
+            first = node;
+            last = node;
+        } else {
+            Node<T> node = new Node<>(value, null, last);
+            last.next = node;
+            last = node;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> newNode = new Node<>(value, first, null);
+            first.prev = newNode;
+            first = newNode;
+            size++;
+            return;
 
+        }
+        Node<T> node = getNode(index);
+        Node<T> newNode = new Node<>(value, node, node.prev);
+        node.prev.next = newNode;
+        node.prev = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
+    }
+
+    private Node<T> getNode(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException(
+                    "This index is not suitable for the given size");
+        }
+        int i = 0;
+        Node<T> node = first;
+        while (i < index) {
+            node = node.next;
+            i++;
+        }
+        return node;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = getNode(index);
+        if (size == 1) {
+
+            first = null;
+            last = null;
+            size--;
+            return node.value;
+        }
+        if (index == size - 1) {
+            node.prev.next = null;
+            last = node.prev;
+            size--;
+            return node.value;
+        }
+        if (index == 0) {
+            node.next.prev = null;
+            first = node.next;
+            size--;
+            return node.value;
+        }
+        node.prev.next = node.next;
+        node.next.prev = node.prev;
+        size--;
```

Comment:
better move all this code to `unlink(Node<T> node)` method and call it here and in the remove(T t)

Reply:
I broke the program ( Give more explanation.

Reply:
You told me left like that.